apps/crn-server/src/data-providers/contentful/analytics.data-provider.ts

```diff
@@ -35,131 +117,48 @@ export class AnalyticsContentfulDataProvider implements AnalyticsDataProvider {
             interestGroupLeadershipRoleCount: team.inactiveSince
               ? 0
               : getUniqueIdCount(
-                  team.linkedFrom?.teamMembershipCollection?.items.flatMap(
-                    (teamMembershipItem) =>
-                      teamMembershipItem?.linkedFrom?.usersCollection?.items
-                        .flatMap(flattenInterestGroupLeaders)
-                        .filter(
-                          (item) =>
-                            item.interestGroupActive &&
-                            item.userIsAlumni === false,
-                        )
-                        .map((item) => item.interestGroupId),
-                  ) || [],
+                  getCurrentInterestGroupIdsFromTeamLeaders(team),
                 ),
-            interestGroupMemberCount: team.inactiveSince
-              ? 0
-              : getUniqueIdCount([
-                  ...(team.linkedFrom?.teamMembershipCollection?.items.flatMap(
-                    (teamMembershipItem) =>
-                      teamMembershipItem?.linkedFrom?.usersCollection?.items
-                        .flatMap(flattenInterestGroupLeaders)
-                        .filter(
-                          (item) =>
-                            item.interestGroupActive &&
-                            item.userIsAlumni === false,
-                        )
-                        .map((item) => item.interestGroupId),
-                  ) || []),
-                  ...(team.linkedFrom?.interestGroupsCollection?.items
-                    .filter((interestGroup) => !!interestGroup?.active)
-                    .flatMap((interestGroup) => interestGroup?.sys.id) || []),
-                ]),
             interestGroupPreviousLeadershipRoleCount: getUniqueIdCount([
-              ...(team.linkedFrom?.teamMembershipCollection?.items.flatMap(
-                (teamMembershipItem) =>
-                  teamMembershipItem?.linkedFrom?.usersCollection?.items
-                    .flatMap(flattenInterestGroupLeaders)
-                    .filter(
-                      (item) => !item.interestGroupActive || item.userIsAlumni,
-                    )
-                    .map((item) => item.interestGroupId),
-              ) || []),
+              ...getPreviousInterestGroupIdsFromTeamLeaders(team),
               ...((team.inactiveSince &&
-                team.linkedFrom?.teamMembershipCollection?.items.flatMap(
-                  (teamMembershipItem) =>
-                    teamMembershipItem?.linkedFrom?.usersCollection?.items
-                      .flatMap(flattenInterestGroupLeaders)
-                      .filter(
-                        (item) =>
-                          item.interestGroupActive &&
-                          item.userIsAlumni === false,
-                      )
-                      .map((item) => item.interestGroupId),
-                )) ||
+                getCurrentInterestGroupIdsFromTeamLeaders(team)) ||
```

Comment:
same as above

Reply:
you would think that wouldn't you - I tried removing it but the issue is that if you use the && expression then if its falsy it returns the value of the first element which is then spread in the array, and since the value of the inactiveSince is not iterable it throws an error, even if we do !!inactiveSince that's still the same case